Source/Core/Core/HW/WiimoteReal/IOWin.cpp

```diff
@@ -690,105 +797,141 @@ int WiimoteWindows::IORead(u8* buf)
-					if (WAIT_TIMEOUT == wait_result)
-					{
-						WARN_LOG(WIIMOTE, "_IOWrite: A timeout occurred on writing to Wiimote.");
-						CancelIo(dev_handle);
-						*written = 0;
-					}
-					else if (WAIT_FAILED == wait_result)
-					{
-						WARN_LOG(WIIMOTE, "_IOWrite: A wait error occurred on writing to Wiimote.");
-						CancelIo(dev_handle);
-						*written = 0;
-					}
-					else if (!GetOverlappedResult(dev_handle, &hid_overlap_write, written, TRUE))
-						*written = 0;
-				}
-				return 1;
-			}
-			else
-			{
-				auto const err = GetLastError();
-				if (ERROR_IO_PENDING == err)
-				{
-					CancelIo(dev_handle);
-				}
-				return 0;
-			}
+	if (written)
+	{
+		if (!GetOverlappedResult(dev_handle, &hid_overlap_write, written, TRUE))
+		{
+			*written = 0;
 		}
 	}
 
+	return 1;
+}
+
+// As of https://msdn.microsoft.com/en-us/library/windows/hardware/ff543402(v=vs.85).aspx, WriteFile is the preferred method
+// to send output reports to the HID. WriteFile sends an IRP_MJ_WRITE to the HID Class Driver (https://msdn.microsoft.com/en-us/library/windows/hardware/ff543402(v=vs.85).aspx).
+// https://msdn.microsoft.com/en-us/library/windows/hardware/ff541027(v=vs.85).aspx & https://msdn.microsoft.com/en-us/library/windows/hardware/ff543402(v=vs.85).aspx
+// state that the used buffer shall be the size of HidCaps.OutputReportSize (the largest output report).
+// However as it seems only the Toshiba Bluetooth Stack, which provides its own HID Class Driver, as well as the HID Class Driver
+// on Windows 7 enforce this requirement. Whereas on Windows 8/8.1/10 the buffer size can be the actual used report size.
+// On Windows 7 when sending a smaller report to the device all bytes of the largest report are sent, which results in
+// an error on the Wiimote. Toshiba Bluetooth Stack in contrast only sends the neccessary bytes of the report to the device.
+// Therefore it is not possible to use WriteFile on Windows 7 to send data to the Wiimote and the fallback
+// to HidP_SetOutputReport is implemented, which in turn does not support "-TR" Wiimotes.
+// As to why on the later Windows' WriteFile or the HID Class Driver doesn't follow the documentation, it may be a bug or a feature.
+// This leads to the following:
+// - Toshiba Bluetooth Stack: Use WriteFile with resized output buffer
+// - Windows Default HID Class: Try WriteFile with actual output buffer (will work in Win8/8.1/10)
+// - When WriteFile fails, fallback to HidP_SetOutputReport (for Win7)
+// Besides the documentation, WriteFile shall be the preferred method to send data, because it seems to use the Bluetooth Interrupt/Data Channel,
+// whereas SetOutputReport uses the Control Channel. This leads to the advantage, that "-TR" Wiimotes work with WriteFile
+// as they don't accept output reports via the Control Channel.
+int _IOWrite(HANDLE &dev_handle, OVERLAPPED &hid_overlap_write, WinWriteMethod &write_method, const u8* buf, size_t len, DWORD* written)
```

Comment:
Function names beginning with an underscore are reserved by the standard for use in the global namespace.